app/controllers/application_controller.rb

```diff
@@ -1615,13 +1615,25 @@ def get_view(db, options = {})
     @targets_hash             = attrs[:targets_hash] if attrs[:targets_hash]
 
     # Set up the grid variables for list view, with exception models below
-    if grid_hash_conditions(view)
+    if grid_hash_conditions(view) && @in_report_data
       @grid_hash = view_to_hash(view)
     end
 
     [view, get_view_pages(dbname, view)]
   end
 
+  # FIXME: This is hotfix for building table and it's settings, however this has to be run only once and if in
+  # report_data.
+  def generate_paged_view_search(view, paged_search_options)
```

Comment:
you don't need to add a test for @in_report_data here, you already have it in the caller do deal with that on the same level.